feat: add uv support for Python package installation

deveworld opened the pull request:
## Summary

Add `--use-uv` flag to use [uv](https://github.com/astral-sh/uv) instead of pip for faster Python package installation.

## Changes

- Add `--use-uv` boolean flag (separate from `--python-choice`)
- Add `check_uv_installed()` function to detect uv availability
- Modify `PYTHON_INSTALL_CMD` to use `uv pip install` when flag is set
- Show warning and ignore flag when used with `pipx` (incompatible)

## Usage

```bash
# With new venv
./install.sh --use-uv

# With active venv
./install.sh --python-choice active-venv --use-uv

# With system python
./install.sh --python-choice system-python --use-uv

# pipx + uv → warning, uv ignored
./install.sh --python-choice pipx --use-uv
```

## Why uv?

- 10-100x faster than pip for package installation
- Drop-in replacement for pip (uv pip install)
- Growing adoption in Python community

## Test result

Tested on Ubuntu 22.04 with Python 3.12 and uv 0.x.x:
- `--help` shows `--use-uv` flag
- `--use-uv` properly sets `PYTHON_INSTALL_CMD="uv pip install"`
- Warning displayed when used with `pipx`

knauth commented:
In the long term, once UV is more of a standard and has better pip compatibility (I've run into uv-specific issues with some of tenstorrent's packages) I'd like it to be the default in tt-installer. I use it on all my dev machines and the UX is indeed much better.

deveworld commented:
Thanks for the feedback! Updated the implementation:
- Changed from `active-uv` option to `--use-uv` flag
- Can be combined with active-venv, new-venv, system-python
- Shows warning and ignores flag when used with pipx
